src/cdomains/addressDomain.ml

```diff
@@ -21,16 +21,32 @@ end
 
 module AddressSet (Idx: IntDomain.Z) =
 struct
-  include Printable.Std (* for default invariant, tag, ... *)
+  module Addr = Lval.NormalLatRepr (Idx)
+  module J = SetDomain.Joined (Addr)
+  (* module H = HoareDomain.SetEM (Addr) *)
+  (* Hoare set for bucket doesn't play well with StrPtr limiting:
+     https://github.com/goblint/analyzer/pull/808 *)
+  include DisjointDomain.ProjectiveSet (Addr) (J) (Addr.R)
 
-  module Addr = Lval.NormalLat (Idx)
-  include HoareDomain.HoarePO (Addr)
+  (* short-circuit with physical equality,
+     makes a different at long-scale: https://github.com/goblint/analyzer/pull/809#issuecomment-1206174751 *)
+  let equal x y = x == y || equal x y
 
   let widen x y =
     if M.tracing then M.traceli "ad" "widen %a %a\n" pretty x pretty y;
     let r = widen x y in
     if M.tracing then M.traceu "ad" "-> %a\n" pretty r;
     r
+  let join x y =
+    if M.tracing then M.traceli "ad" "join %a %a\n" pretty x pretty y;
+    let r = join x y in
+    if M.tracing then M.traceu "ad" "-> %a\n" pretty r;
+    r
+  let leq x y =
+    if M.tracing then M.traceli "ad" "leq %a %a\n" pretty x pretty y;
+    let r = leq x y in
```

Comment:
should `leq` not also short-circuit here?

Reply:
I guess it could, so I added that here.